file_util/src/file.cc

```diff
@@ -0,0 +1,162 @@
+#include <leatherman/file_util/file.hpp>
+#include <boost/nowide/fstream.hpp>
+#include <sstream>
+#include <leatherman/logging/logging.hpp>
+
+using namespace std;
+
+namespace leatherman { namespace file_util {
+
+    bool each_line(string const& path, function<bool(string&)> callback)
+    {
+        boost::nowide::ifstream in(path.c_str());
+        if (!in) {
+            return false;
+        }
+
+        string line;
+        while (getline(in, line)) {
+            if (!callback(line)) {
+                break;
+            }
+        }
+        return true;
+    }
+
+    string read(string const& path)
+    {
+        string contents;
+        if (!read(path, contents)) {
+            return {};
+        }
+        return contents;
+    }
+
+    bool read(string const& path, string& contents)
+    {
+        boost::nowide::ifstream in(path.c_str(), ios::in | ios::binary);
+        ostringstream buffer;
+        if (!in) {
+            return false;
+        }
+        buffer << in.rdbuf();
+        contents = buffer.str();
+        return true;
+    }
+
+    std::string shellExpand(std::string txt) {
+        // This will store the expansion outcome
+        wordexp_t result;
+
+        // Expand and check the success
+        if (wordexp(txt.c_str(), &result, 0) != 0 || result.we_wordc == 0) {
+            wordfree(&result);
+            return "";
+        }
+
+        // Get the expanded text and free the memory
+        std::string expanded_txt { result.we_wordv[0] };
+        wordfree(&result);
+        return expanded_txt;
+    }
+
+    bool fileExists(const std::string& file_path) {
```

Comment:
This seems to be used when setting config files (see cthun-agent/lib/src/configuration.cc) and to trigger some errors (see lib/src/modules/status.cc). Is readability an important part of those tasks? Seems like maybe for the config files it is? Unless of course they would never not be readable...